n_i2c.c

```diff
@@ -297,3 +207,161 @@ bool i2cNoteReset()
     // Done
     return notecardReady;
 }
+
+/**************************************************************************/
+/*!
+  @brief  Receive bytes over I2C to the Notecard.
+  @param   buffer
+            A buffer to receive bytes into.
+  @param   size (in/out)
+            - (in) The size of the buffer in bytes.
+            - (out) The length of the received data in bytes.
+  @param   delay
+            Respect delay standard transmission delays.
+  @param   timeoutMs
+            The maximum amount of time, in milliseconds, to wait for serial data
+            to arrive. Passing zero (0) disables the timeout.
+  @param   overflow (out)
+            The contents did not fit inside the provided buffer.
+  @returns  A c-string with an error, or `NULL` if no error ocurred.
+*/
+/**************************************************************************/
+const char *i2cRawReceive(uint8_t *buffer, size_t *size, bool delay, size_t timeoutMs, size_t *overflow)
+{
+    // Check for special case
+    if (!timeoutMs) {
+        timeoutMs = (size_t)-1;
+    }
+
+    if (delay) {
+        _DelayIO();
+    }
+
+    // Load buffer with chunked I2C values
+    uint32_t available;
+    size_t received = 0;
+    uint16_t requested = 0;
+    *overflow = (received >= *size);
+    const size_t startMs = _GetMs();
+
+    for (bool eop = false ; !(*overflow) ;) {
+        // Read a chunk of data from I2C
+        // The first read will request zero bytes to query the amount of data
+        // available to receive from the Notecard.
+        const char *err = _I2CReceive(_I2CAddress(), &buffer[received], requested, &available);
+        if (err) {
+#ifdef ERRDBG
+            _Debug(err);
+#endif
+            return err;
+        }
+
+        // Add requested bytes to received total
+        received += requested;
+
+        // Request all available bytes, up to the maximum request size
+        requested = (available > 0xFFFF) ? 0xFFFF : available;
+        requested = (requested > _I2CMax()) ? _I2CMax() : requested;
+
+        // Look for end-of-packet marker
+        if (received > 0 && !eop) {
+            eop = (buffer[received-1] == '\n');
+        }
+
+        // Check overflow condition
+        *overflow = ((received + requested) >= *size);
+        if (*overflow) {
+            *overflow = available; // `available` should optimize allocation

```

Comment:
What's meant by the comment here?

Reply:
It's left-over from experimentation. It is a departure from the way we have previously done it. It would allow for allocations larger than `ALLOC_CHUNK`. I'm not sure this is the direction we want to go, because it may cause heap corruption.

I need to have a discussion with Ray, to understand the original intent.